packages/grafana-ui/src/components/Monaco/CodeEditor.story.internal.tsx

```diff
@@ -38,26 +13,41 @@ export default {
     docs: {
       page: mdx,
     },
+    controls: {
+      exclude: ['monacoOptions', 'onEditorDidMount', 'onBlur', 'onSave', 'getSuggestions'],
+    },
   },
-};
+  argTypes: {
+    width: { control: { type: 'range', min: 100, max: 500, step: 10 } },
+    height: { control: { type: 'range', min: 100, max: 800, step: 10 } },
+  },
+} as Meta;
 
-export const basic = () => {
-  const { containerWidth, text, language, showLineNumbers, showMiniMap, readOnly } = getKnobs();
+export const Basic: Story = (args) => {
   return (
     <CodeEditor
-      width={containerWidth}
-      height={400}
-      value={text}
-      language={language}
+      width={args.width}
+      height={args.height}
+      value={args.value}
+      language={args.language}
       onBlur={(text: string) => {
         action('code blur')(text);
       }}
       onSave={(text: string) => {
         action('code saved')(text);
       }}
-      showLineNumbers={showLineNumbers}
-      showMiniMap={showMiniMap}
-      readOnly={readOnly}
+      showLineNumbers={args.showLineNumbers}
+      showMiniMap={args.showMiniMap}
```

Comment:
nit: Changing these props in StoryBook does nothing currently. Not sure if it's an issue in the CodeEditor but I would expect something to change when I click the switches. The readonly switch works :)

Reply:
Yeah...that's weird. Would look into this

Reply:
Did some investigation but couldn't find anything yet

Reply:
Oh...just figured.
I think the issue is from the CodeEditor.

Reply:
Will push a fix now.

Reply:
For the `showMiniMap` option to work, the value needs to be atleast 100 characters long. So, I'm considering more character to it by default

Reply:
Aaah, good catch 🎉

Reply:
Nevermind this, not super important now.